src/platform/silabs/tracing/SilabsPowerTracing.cpp

```diff
@@ -0,0 +1,184 @@
+#include "SilabsPowerTracing.h"
+#include <lib/support/CodeUtils.h>
+#include <platform/silabs/tracing/SilabsTracingConfig.h>
+
+// Need to use the sleeptimer for power tracing. The RAIL timer doesn't tick in EM2 and creates invalid timestamps for this
+// application.
+#ifndef SILABS_GET_SLEEPTIMER_TIME
+#if defined(SL_RAIL_LIB_MULTIPROTOCOL_SUPPORT) && SL_RAIL_LIB_MULTIPROTOCOL_SUPPORT
+#include "sl_sleeptimer.h"
+#define SILABS_GET_SLEEPTIMER_TIME() uint32_t((sl_sleeptimer_get_tick_count64() * 1000ULL) / sl_sleeptimer_get_timer_frequency())
```

Comment:
I think I saw this somewhere else. sSould we just make it an available macro in a header file that can be included when needed?